universal-login-example/src/components/Backup.js

```diff
@@ -1,14 +1,74 @@
 import React, { Component } from 'react';
 import BackupView from '../views/BackupView';
 import PropTypes from 'prop-types';
+import { toWords } from '../Daefen';
+import ethers from 'ethers';
+import DEFAULT_PAYMENT_OPTIONS from '../../config/defaultPaymentOptions';
 
 class Backup extends Component {
+  constructor(props) {
+    super(props);
+    this.identityService = this.props.services.identityService;
+    this.sdk = this.props.services.sdk;
+    this.state = {
+      backupCodes: ['loading...', 'loading...', 'loading...'],
+      btnLabel: 'SET AS BACKUP CODES'
+    };
+  }
+
+  componentDidMount() {
+    this.generateBackupCodes();
+  }
+
+  generateBackupCodes() {
+    var backupCodes = [];
+    for (var i = 0; i < 3; i++) {
+      backupCodes.push(
+        toWords(Math.floor(Math.random() * Math.pow(3456, 4)))
+          .replace(/\s/g, '-')
+          .toLowerCase() +
+          '-' +
+          toWords(Math.floor(Math.random() * Math.pow(3456, 4)))
+            .replace(/\s/g, '-')
+            .toLowerCase()
+      );
+    }
+    this.setState({ backupCodes: backupCodes });
+  }
+
+  async setBackupCodes() {
+    this.setState({ btnLabel: 'Setting...' });
+    const {identityService} = this.props.services;
+    const to = identityService.identity.address;
+    const {privateKey} = identityService.identity;
+    const {sdk} = identityService;
+    var wallets = await Promise.all([
+      ethers.Wallet.fromBrainWallet(identityService.identity.name, this.state.backupCodes[0]), 
+      ethers.Wallet.fromBrainWallet(identityService.identity.name, this.state.backupCodes[1]), 
+      ethers.Wallet.fromBrainWallet(identityService.identity.name, this.state.backupCodes[2])
+    ]);
```

Comment:
I would generate these from the backup code before hand and only show each backup once the key itself was generated. This gives the user a more permanent feeling of "working" to generate these and makes so that the set up process itself is faster. Because it's slower, it gives the user more information on the drawbacks if they want to generate 6 or more.

Reply:
Will make this change